UIPS/UIP-0122.md

```diff
@@ -0,0 +1,93 @@
+---
+uip: "0122"
+title: "%wild: Stateless jet registration"
+description: Propose a new system for deriving jet registrations and hinting which requires no stored state in the interpreter.
+author: ~ritpub-sipsyl (eamsden) 
+status: Draft
+type: Standards Track
+category: Hoon, Nock # Only required for Standards Track. Otherwise, remove this field.
+created: 2024-03-25
+---
+
+## Abstract
+
+This proposal introduces a stateless approach to jet registration in Nock interpreters, utilizing a dynamic hint labeled `%wild`. The `%wild` hint contains the necessary slice of "cold state" for the hinted Nock, thereby avoiding issues with state maintenance and preservation. This approach offers a simple solution to the problems of state in jet registration.
+
+The `%wild` hint can be composed by Hoon and can be implemented using existing information in the subject type, extended only by labels from existing `%fast`-hint runes. It also provides a scoping mechanism which prevents user code from altering kernel code registrations.
+
+An additional advantage of this  approach is its potential to fix the jet mismatch of `+mink`, without requiring any injection of state.
+
+## Motivation
+
+Currently Vere and Ares both handle jet registrations provided by `%fast` hints. By convention and implementation, a `%fast` hint wraps the production of a core, and registers that core's *battery* (axis 2) along with a registered heirarchy of already-registered batteries which match a parent core, whose axis in the newly-registered core is given in the hint. The label provided in a `%fast` hint are consed onto those for parent cores, producing a `$path` for each core registered. This allows us to recognize cores as having a particular label. This ability is fundamental to practical jetting systems.
+
+To correctly implement core labeling with `%fast` hints, an interpreter must maintain *state*. The state directly assembled from `%fast` hints is conventionally referred to as the "cold" state (contrasted with the "hot" state which is shipped with the interpreter binary and maps labels to jets, or the "warm" state which is the join of these two states.)
```

Comment:
this is a good description of cold state -- maybe move it up to the beginning, or say in the beginning that you'll define it later

Reply:
Would it be OK to link to / "incorporate by reference" the ["Fast Hints and Jets"](https://docs.urbit.org/language/nock/guides/jetting) document I wrote a few months ago?